Removed trival warnings

amitsrivastava78 opened the pull request:
Removed some warnings from the files

jdduke commented on tensorflow/lite/arena_planner.cc:
```diff
@@ -153,7 +153,7 @@ TfLiteStatus ArenaPlanner::PlanAllocations() {
+  for (size_t i = 0; i < graph_info_->num_nodes(); ++i) {
     const TfLiteNode& node = graph_info_->node(i);
 
     // First queue output tensors for allocation.
```
I don't think deallocate(i) takes a size_t?

amitsrivastava78 replied:
@jdduke , thanks for pointing this out, i updated the code as per the review comments, kindly check and approve.

Regards
Amit

jdduke commented on tensorflow/lite/kernels/internal/reference/reference_ops.h:
```diff
@@ -2789,7 +2789,7 @@ T FloorMod(T input1, T input2) {
   ModFunc mod_func;
   T trunc_mod = mod_func(input1, input2);
-  return trunc_mod != 0 && (input2 < 0 != trunc_mod < 0) ? trunc_mod + input2
+  return trunc_mod != 0 && ((input2 < 0) != (trunc_mod < 0)) ? trunc_mod + input2
```
Might as well put parens around (trunc_mode != 0) and (trunce_mod + input2)

amitsrivastava78 replied:
@jdduke , thanks for pointing this out, i updated the code as per the review comments, kindly check and approve.

Regards
Amit

Commit:
Fixed warning for the FloorMod.

Removed the warning from the file.

amitsrivastava78 commented:
@jdduke , thanks for pointing this out, i updated the code as per the review comments, kindly check and approve.

Regards
Amit

jdduke commented on tensorflow/lite/arena_planner.cc:
```diff
@@ -153,14 +153,14 @@ TfLiteStatus ArenaPlanner::PlanAllocations() {
   }
   // Go through the graph in execution order.
-  for (int i = 0; i < graph_info_->num_nodes(); ++i) {
+  for (size_t i = 0; i < graph_info_->num_nodes(); ++i) {
```
since j is an int, and we generally use int indices elswhere, should we just keep this as an int and use `static_cast<int>(graph_info_->num_nodes())?

amitsrivastava78 replied:
@jdduke , i have updated the code as per comments, kindly check and update.

Regards
Amit

jdduke commented on tensorflow/lite/arena_planner.cc:
```diff
@@ -284,7 +284,7 @@ TfLiteStatus ArenaPlanner::CalculateTensorDeallocation(int tensor_index) {
 TfLiteStatus ArenaPlanner::CalculateAllocationOfInternalTensors(
     int node_index) {
-  if (node_index < graph_info_->num_nodes()) {
+  if (static_cast<size_t>(node_index) < graph_info_->num_nodes()) {
```
Let's use `static_cast<int>(graph_info_->num_nodes())` for consistency with casts. See also note above.

amitsrivastava78 replied:
@jdduke , i have updated the code as per comments, kindly check and update.

Regards
Amit

amitsrivastava78 commented:
@jdduke , i have updated the code as per comments, kindly check and update.

Regards
Amit

amitsrivastava78 commented:
@jdduke , i found one problem in the function CalculateDeallocationOfInternalTensors and CalculateAllocationOfInternalTensors, this is when the node_index becomes negative(in one of the TC) then dues to the implicit conversion with the unsigned int, negative values were becoming positive and hence the test case was passing, not sure this was intentional or oversight, anyway i changed the code as per your suggestion and added the small error check code to ensure sanity. 
if (node_index < 0) {
return kTfLiteOk;
}
Hope this is ok, Kindly check and approve.

Regards
Amit

amitsrivastava78 commented:
@jdduke , i found one problem in the function CalculateDeallocationOfInternalTensors and CalculateAllocationOfInternalTensors, this is when the node_index becomes negative(in one of the TC) then dues to the implicit conversion with the unsigned int, negative values were becoming positive and hence the test case was passing, not sure this was intentional or oversight, anyway i changed the code as per your suggestion and added the small error check code to ensure sanity.
if (node_index < 0) {
return kTfLiteOk;
}
Hope this is ok, Kindly check and approve.

Regards
Amit

jdduke commented on tensorflow/lite/arena_planner.cc:
```diff
@@ -284,7 +284,10 @@ TfLiteStatus ArenaPlanner::CalculateTensorDeallocation(int tensor_index) {
 TfLiteStatus ArenaPlanner::CalculateAllocationOfInternalTensors(
     int node_index) {
-  if (node_index < graph_info_->num_nodes()) {
+  if (node_index < 0) {
```
It's probably worth a comment as to when node_index would be negative.

amitsrivastava78 replied:
@jdduke thanks for the review, i have updated the comments to describe the problem, kindly check and approve.

Regards
Amit

jdduke commented:
> @jdduke , i found one problem in the function CalculateDeallocationOfInternalTensors and CalculateAllocationOfInternalTensors, this is when the node_index becomes negative(in one of the TC) 

Interesting, can we expand the test to catch this? Can you tell if this would happen for a legit model/interpreter use-case?

amitsrivastava78 commented:
@jdduke this happens when the graph is empty , the TC which is triggering this condition is (ArenaPlannerTest, EmptyGraph)  , which ultimately calls CalculateAllocations which subtracts one from the active node hence causing the problem.

Regards
Amit

jdduke commented:
> @jdduke this happens when the graph is empty , the TC which is triggering this condition is (ArenaPlannerTest, EmptyGraph) , which ultimately calls CalculateAllocations which subtracts one from the active node hence causing the problem.

Can you add a (unit) test for this case?

amitsrivastava78 commented:
@jdduke , thanks for the review i have added one case to cover this scenario, kindly check and approve.

Regards
Amit

jdduke commented on tensorflow/lite/arena_planner_test.cc:
```diff
@@ -210,6 +210,12 @@ TEST_F(ArenaPlannerTest, EmptyGraph) {
   Execute(0, 10);
 }
 
+TEST_F(ArenaPlannerTest, DeallocationOfTensor) {
```
At least a comment here describing what we're testing, it's not at all obvious from the name of the test or the code here.

amitsrivastava78 replied:
@jdduke , sorry for the miss,i have updated the code with the comments, explaining why this TC is added and what problem previously was seen. Hope this is ok.

Regards
Amit

amitsrivastava78 commented:
@jdduke , sorry for the miss,i have updated the code with the comments, explaining why this TC is added and what problem previously was seen. Hope this is ok.

Regards
Amit

jdduke commented on tensorflow/lite/arena_planner.cc:
```diff
@@ -135,7 +135,7 @@ TfLiteStatus ArenaPlanner::PlanAllocations() {
   // Count references to node input tensors.
-  for (size_t i = 0; i < graph_info_->num_nodes(); ++i) {
+  for (int i = 0; i < static_cast<int>(graph_info_->num_nodes()); ++i) {
     const TfLiteNode& node = graph_info_->node(i);
```
Hmm, doesn't node(i) take a size_t?

amitsrivastava78 replied:
@jdduke , thanks for the comments, i have updated the same here and other places as well.

Regards
Amit

jdduke commented on tensorflow/lite/arena_planner.cc:
```diff
@@ -297,7 +297,13 @@ TfLiteStatus ArenaPlanner::CalculateAllocationOfInternalTensors(
+  // For the case if the graph is empty the node index can be negative since we
+  // substract from the active node, so the node_index can be zero for those
+  // cases
+  if (node_index < 0) {
```
Hmm, now I wonder if it would be better to catch this at the callsite in CalculateAllocations? Rather than silently succeeding?

amitsrivastava78 replied:
@jdduke , thanks for the comments, i have shift the check to CalculateAllocations, kindly check.

Regards
Amit

jdduke reviewed (changes requested):
Just a few minor comments, thanks for your patience.

jdduke commented on tensorflow/lite/arena_planner.cc:
```diff
@@ -237,9 +237,14 @@ TfLiteStatus ArenaPlanner::CalculateAllocations(int first_node, int last_node) {
+  // For the case if the graph is empty the node index can be negative since we
+  // substract from the active node, so the node_index can be zero for those
+  // cases
+  if ((active_node - 1) > 0) {
```
Shouldn't this be: active_node > 0? Your condition fails for the active_node == 1 case.

amitsrivastava78 replied:
@jdduke , yes you are right, sorry for the miss, i have updated the code as per your suggestion.
Kindly check.

Regards
Amit

amitsrivastava78 commented:
@jdduke , yes you are right, sorry for the miss, i have updated the code as per your suggestion.
Kindly check.

Regards
Amit

jdduke commented on tensorflow/lite/arena_planner.cc:
```diff
@@ -237,9 +237,14 @@ TfLiteStatus ArenaPlanner::CalculateAllocations(int first_node, int last_node) {
+  // cases
+  if (active_node > 0) {
   // Don't forget to deallocate temporaries of last node.
   TF_LITE_ENSURE_STATUS(
```
Indent here.

amitsrivastava78 replied:
Sure, same is updated in the code.

Regards
Amit

jdduke commented on tensorflow/lite/arena_planner_test.cc:
```diff
@@ -210,6 +210,19 @@ TEST_F(ArenaPlannerTest, EmptyGraph) {
+TEST_F(ArenaPlannerTest, DeallocationOfInputTensor) {
+  // This is a negative TC, which will try to make sure that no allocation for
+  // input tensors is done, hence making it call
+  // CalculateDeallocationOfInternalTensors with negative node_index, since
```
"negative node_index. Since the previous check..."

amitsrivastava78 replied:
Sure, same is updated in the code.

Regards
Amit

jdduke commented on tensorflow/lite/arena_planner_test.cc:
```diff
@@ -210,6 +210,19 @@ TEST_F(ArenaPlannerTest, EmptyGraph) {
+  // CalculateDeallocationOfInternalTensors with negative node_index, since
+  // previous check was doing comparison of node_index which was int and
+  // unsigned int, implicit conversion was passing this case, as the negative
+  // number was converted to unsigned it making it random big int.The new check
```
"was converted to unsigned, making it invalid. The new check..."

amitsrivastava78 replied:
Sure, same is updated in the code.

Regards
Amit

Commit:
Bug Fix and removed trivail warning from the file.

Added a Bug Fix, TC and removed warnings from the file.

amitsrivastava78 commented:
@jdduke , thanks for approving the PR, can you please help to get this merged.

Regards
Amit

jdduke commented:
@rthadur is working to land this.

amitsrivastava78 commented:
> @rthadur is working to land this.

@rthadur any update as to when this can be merged.

Regards
Amit

amitsrivastava78 commented:
@rthadur , can you please help to get this merged.

Regards
Amit

rthadur commented:
Its failing because of internal error .  Here is the error: 
`[==========] Running 14 tests from 1 test suite.
[----------] Global test environment set-up.
[----------] 14 tests from ArenaPlannerTest
[ RUN      ] ArenaPlannerTest.EmptyGraph
[       OK ] ArenaPlannerTest.EmptyGraph (3 ms)
[ RUN      ] ArenaPlannerTest.DeallocationOfInputTensor
Unhandled exception:
    @     0x7feea2d26ce2  GoogleTerminateHandler()
    @     0x7feea2814266  __cxxabiv1::__terminate()
    @     0x7feea2814293  std::terminate()
    @     0x7feea28143cf  __cxa_throw
    @     0x7feea2758f60  std::__g::__throw_out_of_range_fmt()
    @     0x7feea2e33599  tflite::ArenaPlanner::PlanAllocations()
    @     0x7feea47cab82  tflite::(anonymous namespace)::ArenaPlannerTest::SetGraph()
    @     0x7feea47cba8d  tflite::(anonymous namespace)::ArenaPlannerTest_DeallocationOfInputTensor_Test::TestBody()
    @     0x7feea35f960a  testing::Test::Run()
    @     0x7feea35fa973  testing::TestInfo::Run()
    @     0x7feea35fb50b  testing::TestSuite::Run()
    @     0x7feea360ba77  testing::internal::UnitTestImpl::RunAllTests()
    @     0x7feea360b446  testing::UnitTest::Run()
    @     0x7feea47c9701  main
    @     0x7feea23aabbd  __libc_start_main
    @     0x55574cb9e4c9  ../sysdeps/x86_64/start.S:108 _start
terminate called after throwing an instance of 'std::__g::out_of_range'`

amitsrivastava78 commented:
@rthadur , i have fixed the problem.
@jdduke , i know this is old PR, can you please re approve.

Regards
Amit

rthadur commented:
Changes are merged internally , waiting for auto merge to happen.
